### DISCUSSION TITLE
Multi-Vehicle Support

### DISCUSSION THREAD

**Discussion by PlaysWithTrons** [teslamate-org/teslamate#2338] [Q&A]
I recently added a second vehicle to my account (as a secondary user for a family member).

On the one hand, I love that I can see both vehicles in one location, but I hate that my vehicle appears 2nd on the page because my vehicle's name is later in the alphabet.

I wish there was a way to customize the sort, or a way to customize the collapse status of a vehicle on the dashboard pages.

Have others with multiple vehicles found a good solution for this issue?

**Comment by cwanja** [comment]
It is probably second on the page because it is ID number 2 in the database. About the only way I could see fixing that is by dropping the original car (ID 1) and then re-logging in. Not sure if it will add it back as ID 1 or ID 3. I imagine the latter though.

**Reply by PlaysWithTrons** [reply]
I'm afraid I am going to need to display my ignorance here. I am not sure how to check what the current ID assignments are or how to make the ID adjustment you suggest.

It does seem strange to me that the new vehicle, and the one that shows up as the second vehicle on my app, would be assigned a lower ID number - a vehicle I have been capturing data on since July.

**Reply by cwanja** [reply]
No ignorance at all, it is not straight forward. How do you have TeslaMate installed? 

To your point about IDs - it is how relational databases work. The IDs are autogenerated in sequence, 1, 2, 3, etc. They are intended to be the primary key of the table and immutable. Hence why I mentioned removing "ID 1" which should be the vehicle you have been capturing since July and then re-logging in to re-add it to the database. It _should_ come in as ID 3, since ID 2 is assigned to the new vehicle. 

But personally, I would not go to this effort. I would suggest this be a feature to @adriankumpf to be able to reorder the display of vehicles. This should be **worlds** easier than hacking around a database. Does not look like the Issues section has a "feature" request any more, so you should be able to label or change the category of this discussion to an "idea". 

That said, if you still want to know how to check the ID just let me know how you installed TeslaMate and I am happy to suggest how to do so - if I know.

**Reply by PlaysWithTrons** [reply]
It is a docker install on a Synology NAS

**Reply by cwanja** [reply]
Within the directory that you have TeslaMate installed, run the following command:
`docker-compose exec database psql teslamate teslamate -c \ "select id, vin, name from cars;"`

This should return the database ID, Tesla VIN and your Tesla friendly vehicle name. I am assuming that ID 1 is the first vehicle and ID 2 is the second vehicle on TeslaMates homepage. 

**Note**: If you get a blank response, you might need to use `db` instead of `database` depending on what is in your YML file.

**Reply by htc1977** [reply]
instead of deleting a car, you can change the display order within the grafana-json-templates.
Search in all /dashboards/*.json files for
`SELECT name AS __text, id AS __value FROM cars ORDER BY display_priority ASC, name ASC;`
and replace each occurrence with
`SELECT name AS __text, id AS __value FROM cars ORDER BY display_priority ASC, name DESC;`
(twice a file)

That's it.

**Reply by Wenski** [reply]
It would be nice if there is an option to select "favorite" car, so people with many cars or people with a mixed database consisting of old and new ones can be satisfied.

**Reply by YeomansIII** [reply]
A "favorite" car is a great option here. Has there been any more thought on this? I traded in my old Model 3 for a new Y and I now always have to select the new Y from the Grafana variable dropdown. I would like to keep the Model 3 data for historical purposes (which is why we all have this tool anyways ;)

**Reply by olsoybakk** [reply]
@YeomansIII You need to set the `display_priority` for the old car to 2.

> `update cars set display_priority = 2 where id = 1;`

**Reply by olsoybakk** [reply]
Too bad it doesn't change the order in the TeslaMate homepage
